### PR TITLE
backport rustjail systemd cgroup fix #6331 to 3.1

### PR DESCRIPTION
**Pull request by jepio**
This is a backport of #6331 to kata-3.1.

cc @fidencio

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @jepio!

File: src/agent/src/rpc.rs [hunk 1]
```diff
@@ -213,8 +214,13 @@ impl AgentService {
         // determine which cgroup driver to take and then assign to use_systemd_cgroup
         // systemd: "[slice]:[prefix]:[name]"
         // fs: "/path_a/path_b"
+        // If agent is init we can't use systemd cgroup mode, no matter what the host tells us
         let cgroups_path = oci.linux.as_ref().map_or("", |linux| &linux.cgroups_path);
-        let use_systemd_cgroup = SYSTEMD_CGROUP_PATH_FORMAT.is_match(cgroups_path);
+        let use_systemd_cgroup = if self.init_mode {
```
Comment (c3d) [inline src/agent/src/rpc.rs:219]: It's a bit weird to write the condition that way, but OK.

**Review by c3d — APPROVED** [review]
/lgtm, thanks @jepio

**Pull request merged**